src/utils.ts

```diff
@@ -140,17 +136,12 @@ const openSession = async ({
     publicDappKey,
   )
 
-  const sessionTypedData = getSessionTypedData(
-    sessionRequest,
-    await account.getChainId(),
-  )
+  const sessionTypedData = getSessionTypedData(sessionRequest, chainId)
 
-  return useWalletRequestMethods && wallet
-    ? await wallet.request({
-        type: "starknet_signTypedData",
-        params: sessionTypedData,
-      })
-    : await account.signMessage(sessionTypedData)
+  return await wallet.request({
```

Comment:
as discussed with @vladutjs, only wallet rpc spec will be allowed 
starknetkit v2 will be needed